hack/golangci-lint.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o pipefail
+
+# This script exists so that we can ensure that the report files produced by
+# golangci-lint are still copied to the ARTIFACT_DIR (if it exists) even when
+# golangci-lint exits with a non-zero exit code, such as whenever it finds
+# issues.
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+GOTAGS="${1:-""}"
+ARTIFACT_DIR="${ARTIFACT_DIR:-""}"
+
+if [ ! -n "$GOTAGS" ]; then
+  echo "No Go tags provided"
+  exit 1
+fi
+
+cd "$REPO_ROOT"
+
+golangci-lint run --build-tags="$GOTAGS";
+retval="$?"
```

Comment:
The bash equivalent of "try/catch" I've seen is:
```
retval=0
golangci-lint run --build-tags="$GOTAGS" || retval=$?
```

Notably this continues to work with `set -e`.

Reply:
Ahh thanks for the tip! I've seen (and done) this before, but couldn't remember how it interacts with `set -e`. I've incorporated those changes nevertheless.